karabo_bridge/simulation.py

```diff
@@ -107,28 +124,27 @@ def gen_metadata(source, timestamp, trainId):
         return meta
 
     def gen_data(self, trainId):
-        data = dict(self.const_fields)
-
+        data = {}
         timestamp = time()
-        img = self.genfunc()
+        data['image.data'] = self.genfunc()
         base_src = '/'.join((self.source.rpartition('/')[0], '{}CH0:xtdf'))
         sources = [base_src.format(i) for i in range(16)]
-
-        data['__intime'] = timestamp
         # TODO: cellId differ between AGIPD/LPD
-        data['cellId'] = np.arange(self.pulses, dtype=np.uint16)
-        data['combinedFrom'] = sources
-        data['image.data'] = img
-        data['length'] = np.full((self.pulses, 1), img.nbytes, dtype=np.uint32)
-        data['modulesPresents'] = [True for i in range(self.modules)]
-        if self.corrected:
-            data['image.gain'] = np.zeros(self.data_shape, dtype=np.uint16)
-            data['passport'] = self.corr_passport()
-        data['pulseCount'] = self.pulses
+        data['image.cellId'] = np.arange(self.pulses, dtype=np.uint16)
+        if not self.raw:
+            data['image.passport'] = self.corr_passport()
+        if self.modules > 1:
+            # More than one modules have sources
+            data['sources'] = sources
+            data['combinedFrom'] = sources
```

Comment:
is this correct that 'source' and 'combinedFrom' have identical values? doesn't 'combinedFrom' list the devices (output channels) sending the data?

Reply:
Ups, you guys must think I am an idiot! But I just checked and it seems combinedForm is entirely dropped in calib data and only in raw data. I'll change that

Reply:
for raw data it is:
In [19]: data[key]['combinedFrom']                                                                                                                                                                                                            
Out[19]: 
['SPB_DET_AGIPD1M-1/DET/0CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/1CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/2CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/3CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/4CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/5CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/6CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/7CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/8CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/9CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/10CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/11CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/12CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/13CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/14CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/15CH0:xtdf']

In [20]: data[key]['sources']                                                                                                                                                                                                                 
Out[20]: 
['SPB_DET_AGIPD1M-1/DET/0CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/1CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/2CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/3CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/4CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/5CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/6CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/7CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/8CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/9CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/10CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/11CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/12CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/13CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/14CH0:xtdf',
 'SPB_DET_AGIPD1M-1/DET/15CH0:xtdf']

Reply:
but this is only true for raw data